### ISSUE TITLE
build_verify test with riverpod generator

### ISSUE THREAD

**Issue by dipakp2726** [rrousselGit/riverpod#1945]
### Discussed in https://github.com/rrousselGit/riverpod/discussions/1897

<div type='discussions-op-text'>

<sup>Originally posted by **dipakp2726** November 12, 2022</sup>
hello
I am using riverpod_generator and each time I run build runner it changes hash of all providers.
I am using the [build_verify](https://pub.dev/packages/build_verify) test and it fails a lot of times because the provider hash keeps changing which doesn't have anything related to the changes I have made.
any solution for this</div>

## steps
run build verify test when provider generated from code gen

### changes on files:

```
 String $getUserUnreadMessageHash() =>
-    r'6eb53df1b11ba3a3f0034d82afb1151dbd13ae00';
+    r'240039fd433cb59db2dec84f7c0a7fc3d4813cd1';
```

so each time I run I can see similar changes on all providers and this test no longer seems useful at here.
is there any way I can ignore this

**Comment by rrousselGit** [comment]
That's not necessarily an issue. I'd need a complete example

**Comment by dipakp2726** [comment]
reproducible repo,
https://github.com/dipakp2726/json_placeholder_flutter_example

## steps:

 flutter clean
flutter pub get

run the test
the test fails with changes in the providers hash value

run again test:
this time test gets passed.

**Comment by dipakp2726** [comment]
I can see if the test is run twice it works but the build_verify test runs only once and 2nd time it fails because it doesn't clean the workspace/git tree.

https://github.com/kevmoo/build_verify/issues/31

**Comment by dipakp2726** [comment]
HI @rrousselGit 🙋‍♂️

**Comment by rrousselGit** [comment]
I'll admit I'm working on other things atm.
If you fancy it, feel free to investigate.

Riverpod_generator is using a hash mechanism to hash the source of the provider to detect source change.
In theory the output is deterministic, so build_verify shouldn't complain.

**Comment by nohli** [comment]
We have the same effect.

CI shows a diff and a check fails.

Locally, it also shows a diff after first generation. Interestingly, after discarding the diff and generating again, there is no diff.

**Comment by rrousselGit** [comment]
After upgrading the dependencies on the given example, I cannot reproduce the issue.
I'll close this as there's nothing I can do at the moment.


If you still have the problem, please open a new issue with a new example to reproduce the issue.
